File: internal/model/playlist.go
package model

import "database/sql"

type Playlist struct {
	Id            uint64         `json:"id" db:"Id"`
	Name          string         `json:"name" db:"Name"`
	Description   sql.NullString `json:"description" db:"Description"`
	Image         sql.NullString `json:"image" db:"Image"`
	Pos           int64          `json:"order" db:"Pos"`
	CreatedAt     string         `json:"created_at" db:"CreatedAt"`
	PlaylistItems []PlaylistItem `json:"playlist_items"`
}

File: frontend/app/types/playlist.ts
import type { Music } from "./music";

export interface Playlist {
  id: number;
  name: string;
  created_at: string;
  playlist_items: PlaylistItem[];
}

export interface PlaylistItem {
  id: number;
  playlist_id: number;
  music_id: string;
  order_pos: string;
  music: Music;
}
